nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/inheritance/MissingComponentsCheck.java

```diff
@@ -34,15 +35,15 @@ public FlowInheritability checkInheritability(final DataFlow existingFlow, final
         existingMissingComponents.removeAll(proposedFlow.getMissingComponents());
 
         if (existingMissingComponents.size() > 0) {
-            final String missingIds = StringUtils.join(existingMissingComponents, ",");
+            final String missingIds = StringUtils.join(existingMissingComponents.stream().sorted().collect(Collectors.toList()), ",");
```

Comment:
Thanks for the recent updates @ZhewenFu, is the sorting still necessary with the additional changes to use `LinkedHashSet`?

Reply:
Hi, the current fix of using LinkedHashSet only fully solves the first test, and the second test is still flaky so I left the sorting here. I have tried to replace `HashSet` with `LinkedHashSet` on all the places where we check `MissingComponenets` but the second test is still flaky in this case.

Reply:
@ZhewenFu Thanks for the reply. This method also creates two `HashSet` objects, did you try changing them to `LinkedHashSet` as well?